save-demo/src/main/kotlin/com/saveourtool/save/demo/utils/HttpUtils.kt

```diff
@@ -40,26 +46,47 @@ fun ByteReadChannel.toByteBufferFlux(): Flux<ByteBuffer> = DataBufferUtils.readI
  * @param kc [KubernetesClient] that is required for kubernetes interactions
  * @param request callback that receives url and port
  * @return result of type [R]
+ * @throws IOException rethrown from [request]
  */
 suspend fun <R> demoAgentRequestWrapper(
     urlPathSegments: List<String>,
     demoPod: Pod,
     kc: KubernetesClient,
     request: suspend (Url) -> R,
-): R = kc.pods()
-    .withName(demoPod.metadata.name)
-    .portForward(SAVE_DEMO_AGENT_DEFAULT_PORT, Random.nextInt(PORT_RANGE_FROM, PORT_RANGE_TO))
-    .use { portForward ->
-        request(
-            URLBuilder(
-                port = portForward.localPort,
-                pathSegments = urlPathSegments,
-            ).build()
-        )
-            .also {
-                delay(100)
-            }
-    }
+): R {
+    val podName = demoPod.metadata.name
+    val podResource = kc.pods().withName(podName)
+    val containerPort = demoPod.spec.containers.single()
+        .ports
+        .single()
+        .containerPort
+    val saveDemoPort = Random.nextInt(PORT_RANGE_FROM, PORT_RANGE_TO)
```

Comment:
there is a guarantee that port is free? Maybe use `ServerSocket(0).localPort`?

Reply:
For now there is no guarantee, but I reckon we should not improve port-forwarding solution as it is very slow. We should come up with a new solution.